conjureplugin/config/internal/v2/config.go

```diff
@@ -0,0 +1,174 @@
+	Server bool `yaml:"server,omitempty"`
+	// CLI indicates if we will generate cobra CLI bindings. Currently this is behind a feature flag and is subject to change.
+	CLI bool `yaml:"cli,omitempty"`
+	// AcceptFuncs indicates if we will generate lambda based visitor code.
+	// Currently this is behind a feature flag and is subject to change.
+	AcceptFuncs *bool `yaml:"accept-funcs,omitempty"`
+	// Extensions contain metadata for consumption by assets of type `conjure-ir-extensions-provider`.
+	Extensions map[string]any `yaml:"extensions,omitempty"`
+	// OmitTopLevelProjectDir skips creating the {ProjectName} subdirectory.
+	// When false (default), generates into {OutputDir}/{ProjectName}/.
+	// When true, generates directly into {OutputDir}/.
+	OmitTopLevelProjectDir bool `yaml:"omit-top-level-project-dir,omitempty"`
+	// SkipDeleteGeneratedFiles skips cleanup of old generated files before regeneration.
+	// When false (default), deletes all Conjure-generated files in the output directory tree before regenerating.
+	// When true, preserves v1 behavior (no cleanup).
+	SkipDeleteGeneratedFiles bool `yaml:"skip-delete-generated-files,omitempty"`
+}
+
+// OutputDirConflicts detects output directory conflicts between projects.
+// Returns a map from project name to a list of errors describing conflicts with other projects.
+//
+// Two types of conflicts are detected:
+//   - Identical output directories: Two projects writing to the exact same directory
+//   - Nested output directories: One project's output directory is a subdirectory of another's
+//
+// Projects are compared in sorted order to ensure deterministic error messages.
+// An empty map indicates no conflicts were found.
+// For identical directories, only one error is reported (for the lexicographically first project).
+func (c *ConjurePluginConfig) OutputDirConflicts() map[string][]error {
+	type Project struct {
+		name   string
+		config SingleConjureConfig
+	}
+
+	var projects []Project
+	for project, config := range c.ProjectConfigs {
+		projects = append(projects, Project{
+			name:   project,
+			config: config,
+		})
+	}
+
+	sort.Slice(projects, func(i, j int) bool {
+		return projects[i].name < projects[j].name
+	})
+
+	result := make(map[string][]error)
+	for i1, p1 := range projects {
+		for i2, p2 := range projects {
+			if i1 == i2 {
+				continue
+			}
+
+			p1Dir := p1.config.ResolvedOutputDir(p1.name)
+			p2Dir := p2.config.ResolvedOutputDir(p2.name)
+
+			if p1Dir == p2Dir && i1 < i2 {
+				// For identical directories, report the error for the first project only
+				result[p1.name] = append(result[p1.name], fmt.Errorf("project %qs output directory %q is the same as project %qs output directory", p1.name, p1Dir, p2.name))
+			} else if validate.IsSubdirectory(p1Dir, p2Dir) {
+				result[p1.name] = append(result[p1.name], fmt.Errorf("project %qs output directory %q contains project %qs output directory %q as a subdirectory", p1.name, p1Dir, p2.name, p2Dir))
+			}
+		}
```

Comment:
Update error output so that we don't have `%qs`, which is awkward to read.

```
fmt.Errorf("project %q and %q have the same output directory %q", p1.name, p2.name, p1Dir)
```

```
fmt.Errorf("output directory %q of project %q is a subdirectory of output directory %q of project %q" ...
```